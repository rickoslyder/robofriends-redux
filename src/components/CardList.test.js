import { render, screen } from '@testing-library/react';
import React from 'react';
import CardList from './CardList';

const mockRobots = [
    {
        name: "test1",
        email: "test1@test.com",
        id: "1"
    },
    {
        name: "test2",
        email: "test2@test.com",
        id: "2"
    },
    {
        name: "test3",
        email: "test3@test.com",
        id: "3"
    },
]

const mockCardListComponent = <CardList robots={mockRobots} />

test('expect to render CardList component', () => {
    render(mockCardListComponent)
    const headers = screen.getAllByRole("heading")
    expect(headers).toBeDefined();
    expect(headers.length).toBeGreaterThanOrEqual(1)
})

test('expect mock Card component JSX to match snapshot', () => {
    expect(mockCardListComponent).toMatchSnapshot()
})

test('expect rendered mock Card component to match snapshot', () => {
    expect(render(mockCardListComponent).baseElement).toMatchSnapshot()
})