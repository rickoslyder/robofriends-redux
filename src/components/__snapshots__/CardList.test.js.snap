// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect mock Card component JSX to match snapshot 1`] = `
<CardList
  robots={
    Array [
      Object {
        "email": "test1@test.com",
        "id": "1",
        "name": "test1",
      },
      Object {
        "email": "test2@test.com",
        "id": "2",
        "name": "test2",
      },
      Object {
        "email": "test3@test.com",
        "id": "3",
        "name": "test3",
      },
    ]
  }
/>
`;

exports[`expect rendered mock Card component to match snapshot 1`] = `
<body>
  <div>
    <div>
      <div
        class="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
      >
        <img
          alt="robots"
          src="https://robohash.org/1?size=200x200"
        />
        <div>
          <h2>
            test1
          </h2>
          <p>
            test1@test.com
          </p>
        </div>
      </div>
      <div
        class="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
      >
        <img
          alt="robots"
          src="https://robohash.org/2?size=200x200"
        />
        <div>
          <h2>
            test2
          </h2>
          <p>
            test2@test.com
          </p>
        </div>
      </div>
      <div
        class="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
      >
        <img
          alt="robots"
          src="https://robohash.org/3?size=200x200"
        />
        <div>
          <h2>
            test3
          </h2>
          <p>
            test3@test.com
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
